vm/map.c

```diff
@@ -775,11 +775,29 @@ void vm_mapDestroy(process_t *p, vm_map_t *map)
 
 	proc_lockDone(&map->lock);
 #else
-	proc_lockSet(&map->lock);
-	while ((e = p->entries) != NULL) {
-		_map_remove(map, e);
-		map_free(e);
+	map_entry_t *next;
+
+	proc_lockSet2(&map->lock, &p->lock);
+	if (p->entries != NULL) {
+		e = p->entries;
+		do {
+			next = e->next;
+			/* Remove only entries associated with the map */
+			if (e->map == map) {
+				LIST_REMOVE(&p->entries, e);
+				if (next == e) {
+					next = NULL;
+				}
+				e->process = NULL;
+				lib_rbRemove(&map->tree, &e->linkage);
+				e->map = NULL;
+				map_free(e);
```

Comment:
`map_common.lock` is locked and unlocked here every time, maybe number of locking may be reduced?

Reply:
Done